controllers/auth.js

```diff
@@ -144,11 +144,28 @@ exports.register = asyncHandler(async (req, res, next) => {
   }
 
   // Get reset token
-  const resetToken = generateOTP(6);
-  user.resetPasswordToken = resetToken;
-  await user.save({ validateBeforeSave: false });
+   const token = jwt.sign({ id: user._id }, process.env.JWT_SECRET, {
+     JWT_SECRET: '5m'
+   });
+
+   user.resetPasswordToken = resetToken;
+   if (resetToken) {
```

Comment:
We Are not handling the reset password here 

Imaging Send a token with this routes 

and awaiting another response with this same route to issue new JWT creds 


This Is meant to be handled in a different endpoint 

* Forget Password
* generate email 
* enter OTP with email 
* Find the user and check the time (Timeout logic )
* send new JWT creds 
* updated password 
* End 

this step requires three different route